### PR TITLE
Treat balance task with released source segment as stale

### LINKED ISSUE #26444: [Bug]: A segment is balanced to two nodes at the same time
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

image
image
And there are many repeated tasks in a segment.

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by yah01**
/kind improvement
fix #26444

### CODE REVIEW

File: internal/querycoordv2/task/scheduler.go [hunk 1]
```diff
@@ -779,8 +779,16 @@ func (scheduler *taskScheduler) checkSegmentTaskStale(task *SegmentTask) error {
 			}
 
 		case ActionTypeReduce:
-			// Do nothing here,
-			// the task should succeed if the segment not exists
+			// Check for balance task only,
+			// the release task should succeed if the segment not exists.
+			if GetTaskType(task) == TaskTypeMove {
+				leaderSegmentDist := scheduler.distMgr.LeaderViewManager.GetSealedSegmentDist(task.SegmentID())
```
Comment (jiaoew1991) [inline internal/querycoordv2/task/scheduler.go:785]: add more ut here

**Comment by yah01** [comment]
rerun ut

File: internal/querycoordv2/task/scheduler.go [hunk 2]
```diff
@@ -804,8 +828,14 @@ func (scheduler *taskScheduler) checkChannelTaskStale(task *ChannelTask) error {
...
 
 		case ActionTypeReduce:
-			// Do nothing here,
-			// the task should succeed if the channel not exists
+			// Check for balance task only,
+			// the unsubscribe task should succeed if the channel not exists.
+			if GetTaskType(task) == TaskTypeMove {
+				channelDist := scheduler.distMgr.LeaderViewManager.GetChannelDist(task.Channel())
+				if !lo.Contains(channelDist, task.Actions()[1].Node()) {
+					return merr.WrapErrServiceInternal("source segment released, stop balancing")
```
Comment (congqixia) [inline internal/querycoordv2/task/scheduler.go:836]: source channel here, not segment

**Review by congqixia — COMMENTED** [review]
/cc @xiaofan

**Review by congqixia — COMMENTED** [review]
/lgtm

**Review by congqixia — COMMENTED** [review]
/approve

**Pull request merged**